src/DripsHub.sol

```diff
@@ -151,46 +152,48 @@ abstract contract DripsHub {
         uint128 amt
     );
 
-    struct ReceiverState {
-        // The amount collectable independently from cycles
-        uint128 collectable;
+    struct UserOrAccount {
+        bool isAccount;
+        address user;
+        uint256 account;
+    }
+
+    struct DripsHubStorage {
+        /// @notice User drips states.
+        /// The keys are the user hash and the asset ID.
+        mapping(bytes32 => mapping(uint256 => DripsState)) dripsStates;
```

Comment:
Why is the asset ID a uint and not an address?

I mean technically it is anyway the hash of it, so it doesn't really matter.

Reply:
On the abstract DripsHub level the token address doesn't really matter, it's just a mapping key and an event field, so it could be anything capable of telling apart 2^20 values. If we want to go in the library direction, we'll need to make it as universal as possible and not assume that the asset is an ERC-20 token identified by its address, this is the role of the wrapping contract implementing a specific API. I think that in this context uint256 will be the right choice.

So should we stick with `uint256`, tighten it down to `address` or make even bigger assumptions and make it `IERC20`? I think that `uint256` is both the cleanest and the most aligned with our future plans preventing disruptions. WDYT?

Reply:
I like the idea of abstracting it more by using a `uint256`. This would allow other use cases e.g. for games that use a unique point system or for applications that are similar to how current real world reward systems work that don't want to have a tradable token.

I can also imagine that this would be useful for a later stage when we want to add anonymity to the protocol.